src/main/kotlin/me/aurieh/ichigo/core/CommandContext.kt

```diff
@@ -48,6 +49,14 @@ class CommandContext internal constructor(val receivedEvent: MessageReceivedEven
         }
     }
 
+    inline fun<reified T : Event> waitFor(crossinline predicate: (T) -> Boolean, timeout: Int = -1): T {
```

Comment:
should be suspend as well, though I will add it soon:tm:

Reply:
Any progress on this yet?